katdal/h5datav2_5.py

```diff
@@ -0,0 +1,631 @@
+        returned as double-precision floating points.
+        The shape of the array is (*T*, *F*, *B*), with time along the first
+        dimension, frequency along the second dimension. The third dimension
+        consists of LL and RR autocorrelation.
+
+        The returned array always has all three dimensions,
+        even for scalar (single) values. The number of integrations *T* matches
+        the length of :meth:`timestamps`, the number of frequency channels *F*
+        matches the length of :meth:`freqs` and the number of correlation
+        products *B* matches the length of :meth:`corr_products`. To get the
+        data array itself from the indexer `x`, do `x[:]` or perform any other
+        form of indexing on it. Only then will data be loaded into memory.
+        """
+        extract = LazyTransform('extract_vis',
+                                lambda vis, keep: vis.astype(np.float32),
+                                lambda shape: shape, np.float32)
+        return self._vislike_indexer(self._vis, extract)
+
+    @property
+    def stokes(self):
+        """Single-dish observational data, 32-bit signed integer, Stokes Q and U. Same
+        (*T*, *F*, *B*) arrangement as the vis data.
+        """
+        extract = LazyTransform('extract_stokes',
+                                lambda stokes, keep: stokes.astype(np.float32),
+                                lambda shape: shape, np.float32)
+        return self._vislike_indexer(self._stokes, extract)
+
+    @property
+    def time_av_ll(self):
+        def _extract_time_av_ll(ll, keep):
+            return ll.astype(np.float64)
+        extract_time_av_ll = LazyTransform('extract_time_av_ll', _extract_time_av_ll, lambda shape: shape, np.float64)
+        return LazyIndexer(self._time_av_ll, transforms=[extract_time_av_ll])
+
+    @property
+    def time_av_rr(self):
+        def _extract_time_av_rr(rr, keep):
+            return rr.astype(np.float64)
+        extract_time_av_rr = LazyTransform('extract_time_av_rr', _extract_time_av_rr, lambda shape: shape, np.float64)
+        return LazyIndexer(self._time_av_rr, transforms=[extract_time_av_rr])
+
+    @property
+    def time_av_q(self):
+        def _extract_time_av_q(q, keep):
+            return q.astype(np.float64)
+        extract_time_av_q = LazyTransform('extract_time_av_q', _extract_time_av_q, lambda shape: shape, np.float64)
+        return LazyIndexer(self._time_av_q, transforms=[extract_time_av_q])
+
+    @property
+    def time_av_u(self):
+        def _extract_time_av_u(u, keep):
+            return u.astype(np.float64)
+        extract_time_av_u = LazyTransform('extract_time_av_u', _extract_time_av_u, lambda shape: shape, np.float64)
+        return LazyIndexer(self._time_av_u, transforms=[extract_time_av_u])
+
+    @property
+    def u(self):
+        raise NotImplementedError("File is AVN Single-dish format, uv-plane is not applicable.")
```

Comment:
Or you could just return 0 :-)

Reply:
Would that be valid? I'm concerned that if a user unwittingly uses this, it may cause unexpected weird results.

Reply:
It should be. That's where single-dish data goes...

Reply:
Okay. Would that be true of u, v and w? Or just u?

Reply:
All of them.

Reply:
Done in ca0e58d. Have I understood correctly?